MoreLinq.Test/ToLookupTest.cs

```diff
@@ -0,0 +1,118 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008 Jonathan Skeet. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class ToLookupTest
+    {
+        [Test]
+        public void ToLookupWithNullKeyValuePairSequence()
+        {
+            Assert.ThrowsArgumentNullException("source", () =>
+                MoreEnumerable.ToLookup((IEnumerable<KeyValuePair<object, object>>) null));
```

Comment:
> Conceptually, I find it wrong to test behavior from the underlying library

You could also look at it as testing an assumption. As long as the test never went red initially, there was nothing to fix!

> What if in some version of the runtime the first argument to `ToLookup` is not named "source"?

Then the tests would break and we'll know!

And while you raise a fair point, I've realized that these tests are now redundant with [`NullArgumentTest`](https://github.com/morelinq/MoreLINQ/blob/b25eef043c6a54adead8428dcf2db5cf23f2dd6d/MoreLinq.Test/NullArgumentTest.cs). Perhaps all such tests should be removed? :thinking: